### DISCUSSION TITLE
How can I edit release message?

### DISCUSSION THREAD

**Discussion by avatar-lavventura** [magit/magit#4927] [Support] [answered]
In GitHub under releases tag we can edit into already released tag at `https://github.com/<domain>/<name>/releases/edit/<tag_version>` and update release does the job.

<img width="811" alt="My Screenshot 2023-04-21 at 20 50 10" src="https://user-images.githubusercontent.com/18537398/233702261-111a662b-2014-4f80-89a9-cb37d18a65cb.png">

This could also be done through `cli.github` https://cli.github.com/manual/gh_release_edit

I was wondering can we achieve this using `magit`?

PS: magit is a coolest interface in emacs.

**Comment by kyleam** [comment, marked as answer]
No, Magit doesn't have any support for editing [GitHub releases][gr], which is a GitHub-specific feature built on top of regular Git tags.

> In GitHub under releases tag we can edit into already released tag

In that case, you're editing the GitHub release message, not the tag itself.

If you want to edit the message of an annotated tag, the only option is to replace that tag with another tag that has the same name and a different message.  For example, you can call `tag` with `--annotate` and `--force` (`t -a -f`) and specify the name of an existing tag.  When an annotated tag with that name exists, the message buffer that pops up will be contain the old message for you to edit.

[gr]: https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases

**Comment by avatar-lavventura** [comment]
> The only option is to replace that tag with another tag that has the same name and a different message.

As I understand, we cannot replace tag with the same tag, or delete that tag and re-create the it using deleted tag's version with updated message.

=> Can we achieve  replace that tag with another tag using magit? Basic example would be helpful.

Thank you for your help.

**Reply by kyleam** [reply]
> As I understand, we cannot replace tag with the same tag, or delete that tag and re-create the it using deleted tag's version with updated message.

You can.

```
$ git tag -a -m 'tag message' v1

$ git for-each-ref refs/tags
9ec2d589221313f2e9815cb82b4e2252fd5db694 tag	refs/tags/v1

$ git cat-file tag refs/tags/v1
object e2a9ac31c1663aaa8de8a56d4f6a805b1b791113
type commit
tag v1
tagger Kyle Meyer <kyle@kyleam.com> 1682381487 -0400

tag message

# can't overwrite existing tag by default...
$ git tag -a -m 'new message' v1
fatal: tag 'v1' already exists
# ... but can if you add --force

$ git tag -a -f -m 'new message' v1
Updated tag 'v1' (was 9ec2d58)

$ git for-each-ref refs/tags
a26a2ec013f318a0e398cc958a9000d1ba7cdcba tag	refs/tags/v1

$ git cat-file tag refs/tags/v1
object e2a9ac31c1663aaa8de8a56d4f6a805b1b791113
type commit
tag v1
tagger Kyle Meyer <kyle@kyleam.com> 1682381542 -0400

new message
```

You could also do that in two steps: 1) delete tag and 2) re-tag (no `--force` needed).  The main advantage of `--force` is that, if you don't supply the message with `-m`, Git pops up the old message for you to edit.

> => Can we achieve replace that tag with another tag using magit? Basic example would be helpful.

The `magit-tag` transient has a `--force` argument.  So, to force create an annotated tag, you can use the sequence I gave in my original reply: `t -a -f`.

**Reply by avatar-lavventura** [reply]
@kyleam Thanks!
